pydevd.py

```diff
@@ -1077,8 +1077,9 @@ def get_file_type(self, frame, abs_real_path_and_basename=None, _cache_file_type
             if abs_real_path_and_basename[0] == "<string>":
                 # Consider it an untraceable file unless there's no back frame (ignoring
                 # internal files and runpy.py).
-                if frame.f_back is not None and self.get_file_type(frame.f_back) == self.PYDEV_FILE:
-                    # Special case, this is a string coming from pydevd itself (or another internal file)
+                if frame.f_back is not None and self.get_file_type(frame.f_back) == self.PYDEV_FILE and is_pydevd_path(frame.f_back.f_code.co_filename):
```

Comment:
This is the change required to let '<string>' modules from debugpy to go through. Debugpy modules are marked as PYDEV_FILES but the code they run should be marked as NONE. Only the lambdas in pydevd should be marked as PYDEV_FILE.

Reply:
I must say I don't fully comprehend the use case... can you elaborate a bit on which use-case this happens? Maybe `debugpy` itself could not use `<string>` as the name to compile those in that use case?

Reply:
I believe it was debugpy using runpy to execute some string code. I'll have to pull this back and see which tests fail.

Reply:
It's this test here.
https://github.com/microsoft/debugpy/blob/8ab4ee89e9ae9a926f711904900a27e3c96d43d4/tests/debugpy/test_breakpoints.py#L420

It's whenever debugpy is launched with a string. That string's caller is debugpy so it goes down this path. 

I'll look into maybe making pydevd's strings to be marked special instead. Then we can eliminate this code.